delivery_price_by_category/models/delivery_carrier.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+from odoo.tools import safe_eval
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = 'delivery.carrier'
+
+    @api.multi
+    def get_price_available(self, order):
```

Comment:
```suggestion
    @api.multi
    def get_price_available(self, order):
        self.ensure_one()
        category_price = 0.0
        price_dict = self.get_price_dict(order)
        for line in self.price_rule_ids:
            if line.product_category_id:
                products = order.mapped('order_line.product_id')
                if any(product.categ_id == line.product_category_id for product in products):
                    category_price = line.product_category_price
                    break
        if category_price:
            return category_price
         # Note that this will evaluate all the price_rule_ids again and
        # our category rules might interfere withthe correct computation
        return super(DeliveryCarrier, self).get_price_available(order)
```

Your tests on the else part can be removed. Keep it simple :smile: 

And I think the loop can be avoided if (but I don't know the process) if you find directly the category price

Reply:
@rousseldenis thanks for the code suggestion! 
The test in the `else` part is necessary because if the rules are:

1. standard rule that will pass
2. category rule that will pass

your method would apply the price of rule 2 but the expected behavior is to have the price of rule 1

Reply:
Sorry, mixed break and continue... end of day...

Reply:
@rousseldenis ahah yes after 8 hours is better to `break` than to `continue`..

Anyway, what I said in https://github.com/OCA/delivery-carrier/pull/165#discussion_r231609393 is not resolved with a `continue`: I think we should always check both types of rule in order to get the 1st one which satisfies the condition (`safe_eval` for standard rules and `category` for _category_ rules).

Too bad the core method is not really written to be easily overridden.